### ISSUE TITLE
btrfs-dedupe serialization error

### ISSUE THREAD

**Issue by a-schild** [jamespharaoh/rust-btrfs#1]
Hello,

I installed the btrfs-dedupe from the ubuntu packages at https://dist.wellbehavedsoftware.com/btrfs-dedupe/ for xenial.

Running `/usr/local/sbin/btrfs-dedupe dedupe --database /var/cache/btrfs-dedupe/Data.gz /Data`
aborts with this message:

```
Scanning /Data
Scanned 14208673 files
Total 12277197 files in database
Serialization error: invalid value: Path contains invalid UTF-8 characters at line 0 column 0
```
It seems that it struggels over this filename which apparently has another encoding, which then causes dedupe to fail

`-rw-r--r-- 1 root root 72 Mär 10  2004 ?slenska.alias
`

**Comment by jamespharaoh** [comment]
Yeah, I am assuming a utf-8 encoding. There are various ways this could be dealt with, but it's not a use case I was particularly interested in when I built it.

I guess the solution would involve storing path names in an open 8-bit format, possibly as an option, and possibly with an option for the default encoding, and/or looking at the environment.

I am not currently able to release new code due to a new employment contract. If you want to create a pull request that's fine. If not, I also would like to fix this at some point, and I believe I should be able to continue to work on this project at some point.
